lib/durationPicker.dart

```diff
@@ -0,0 +1,649 @@
+              onPressed: _handleCancel),
+          new FlatButton(
+              child: new Text(localizations.okButtonLabel), onPressed: _handleOk),
+        ]));
+
+    final Dialog dialog = new Dialog(child: new OrientationBuilder(
+        builder: (BuildContext context, Orientation orientation) {
+          final Widget pickerAndActions = new Container(
+            color: theme.dialogBackgroundColor,
+            child: new Column(
+              mainAxisSize: MainAxisSize.min,
+              children: <Widget>[
+                new Expanded(
+                    child:
+                    picker), // picker grows and shrinks with the available space
+                actions,
+              ],
+            ),
+          );
+
+          assert(orientation != null);
+          switch (orientation) {
+            case Orientation.portrait:
+              return new SizedBox(
+                  width: _kDurationPickerWidthPortrait,
+                  height: _kDurationPickerHeightPortrait,
+                  child: new Column(
+                      mainAxisSize: MainAxisSize.min,
+                      crossAxisAlignment: CrossAxisAlignment.stretch,
+                      children: <Widget>[
+                        new Expanded(
+                          child: pickerAndActions,
+                        ),
+                      ]));
+            case Orientation.landscape:
+              return new SizedBox(
+                  width: _kDurationPickerWidthLandscape,
+                  height: _kDurationPickerHeightLandscape,
+                  child: new Row(
+                      mainAxisSize: MainAxisSize.min,
+                      crossAxisAlignment: CrossAxisAlignment.stretch,
+                      children: <Widget>[
+                        new Flexible(
+                          child: pickerAndActions,
+                        ),
+                      ]));
+          }
+          return null;
+        }));
+
+    return new Theme(
+      data: theme.copyWith(
+        dialogBackgroundColor: Colors.transparent,
+      ),
+      child: dialog,
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+}
+
+/// Shows a dialog containing the duration picker.
+///
+/// The returned Future resolves to the duration selected by the user when the user
+/// closes the dialog. If the user cancels the dialog, null is returned.
+///
+/// To show a dialog with [initialTime] equal to the current time:
+///
+/// ```dart
+/// showDurationPicker(
+///   initialTime: new Duration.now(),
+///   context: context,
+/// );
+/// ```
+Future<Duration> showDurationPicker(
+  {@required BuildContext context,
+  @required Duration initialTime,
+  double snapToMins}) async {
+  assert(context != null);
+  assert(initialTime != null);
+
+  return await showDialog<Duration>(
+    context: context,
+    builder: (BuildContext context) =>
+    new _DurationPickerDialog(initialTime: initialTime, snapToMins: snapToMins),
+  );
+}
```

Comment:
We don't use this part of the package (or at the very least, our app doesn't seem to break when I delete it). In order to make it easier to scan and read for everybody else, I'd omit it.